RocketChat/Rocket.Chat
# fix: Generic Table Search

## Proposed changes (including videos or screenshots)

When the user changes the page and tries to search for a room that is not currently visible on the page it’s not returning any results. This is also reproducible in the Users table 

**Current Behavior:**
It returns the list filtering by the name
![task_table](https://user-images.githubusercontent.com/20212776/224804926-d8bffe6e-419f-444f-bdfa-ede01073527d.gif)


**Previous Behavior:**
Not found - It was not finding the room because it’s filtering only in the current page
![search-pagination-rooms](https://user-images.githubusercontent.com/20212776/224803475-e991b354-df3d-42fe-8564-b041ca5edc89.gif)





## Issue(s)


## Steps to test or reproduce


1. Go to Admin → Rooms list
2. Go to the last page possible
3. Search for a room that is shown on the first page

## Further comments

TC-524

I also added the `refetchOnWindowFocus: false,` to useQuery methods to avoid re-fetching data on user focus on the page, this avoid a lot requests
